File: Programmers/Level1/크레인 인형뽑기 게임.py
def solution(board, moves):
    answer = 0
    newboard = [[0]*len(board) for _ in range(len(board))]
    busket = []
    for i in range(len(board)):
        for j in range(len(board)):
            newboard[i][j] = board[j][i]

    for m in moves:
        for i in range(len(newboard[m-1])):
            if(newboard[m-1][i] == 0):
                continue
            else:
                if(busket and newboard[m-1][i] == busket[-1]):
                    del busket[-1]
                    answer += 2
                    newboard[m-1][i] = 0
                    break
                else:
                    busket.append(newboard[m-1][i])
                    newboard[m-1][i] = 0
                    break

    return answer

File: Swift 코딩테스트 연습/크레인 인형뽑기게임.swift
import Foundation

func solution(_ board:[[Int]], _ moves:[Int]) -> Int {
    var newBoard = Array(repeating: Array(repeating: 0, count: board.count), count: board.count) 
    var busket: [Int] = []
    var count:Int = 0

    for i in 0..<board.count {
        for j in 0..<board.count {
            newBoard[j][i] = board[i][j]
        }
    }
    
    for i in 0..<board.count {
        newBoard[i].reverse()
    }
    
    for i in 0..<moves.count {
        while !newBoard[moves[i] - 1].isEmpty {
            var doll = newBoard[moves[i] - 1].removeLast()
            if doll != 0 {
                if !busket.isEmpty && busket.last == doll {
                    busket.removeLast()
                    count += 2
                } else {
                    busket.append(doll)
                }
                break
            }
        }  
    }
    
    return count
}